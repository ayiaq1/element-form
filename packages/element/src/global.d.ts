/*
 * @Author: yihuang
 * @Date: 2021-07-21 11:16:16
 * @Description: 推导格式
 * @LastModifiedBy: yihuang
 */
declare module '*.svg';
declare module '*.jpg';
declare module '*.png';

declare module '*.module.css' {
  export const content: Record<string, string>;
  export default content;
}
declare module '*.module.less' {
  export const content: Record<string, string>;
  export default content;
}
