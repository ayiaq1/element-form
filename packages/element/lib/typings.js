/** Unix 时间戳（自 Unix 纪元以来的秒数） */
export var Unix = 'unix';
/** Unix 毫秒时间戳（自 Unix 纪元以来的毫秒数） */
export var UnixValueOf = 'unixValueOf';
/** 返回 Moment 格式 */
export var MOMENT = 'moment';
export var HMS = 'HH:mm:ss';
export var HM = 'HH:mm';
export var EnumDataType = {
  /** HH:mm:ss */
  HMS: HMS,
  /** YYYY:HH:mm:ss */
  YMD: 'YYYY-MM-DD',
  /** YYYY-MM-DD HH:mm:ss */
  YMD_HMS: 'YYYY-MM-DD HH:mm:ss',
  /** YYYY-MM-DD 00:00:00 */
  YMD_START_TIME: 'YYYY-MM-DD 00:00:00',
  /** YYYY-MM-DD 23:59:59 */
  YMD_END_TIME: 'YYYY-MM-DD 23:59:59',
};
/** 给 外部判断type用 */
export var Date_Picker = 'datePicker';
export var Range_Picker = 'rangePicker';
export var Time_Picker = 'timePicker';
export var Time_Range_Picker = 'timeRangePicker';
