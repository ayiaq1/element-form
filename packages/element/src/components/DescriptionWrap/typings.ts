import { IBaseFieldProps } from '../../typings';
import { DescriptionsProps } from 'antd';

export interface IDescOption {
  /** 标题 */
  label: string;
  /** 字段名 */
  value: string;
  /** 对应的key，一般给远程加载使用 */
  code: string;
}
export interface IUrlMap {
  url: string;
  method: 'post' | 'get';
  params?: any;
}
export interface IDescProps {
  /** 是否需要查询 */
  urlMap?: IUrlMap;
  options?: IDescOption[];
}

export type IDesFieldProps = IBaseFieldProps & DescriptionsProps & IDescProps;
