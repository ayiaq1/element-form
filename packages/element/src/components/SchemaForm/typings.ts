import { FormInstance, FormItemProps } from 'antd/lib/form';
import { CSSProperties } from 'react';
import {
  IText,
  IInputProps,
  ICustom,
  IInput,
  ITextProps,
  ISelectProps,
  ISelect,
  IDatePicker,
  IDatePickerProps,
  IRangePicker,
  IRangePickerProps,
  ICheckbox,
  ICheckboxProps,
  ICheckboxGroup,
  ICheckboxGroupProps,
  ITextArea,
  ITextAreaProps,
  ISearchProps,
  ISearch,
  IPassword,
  IPasswordProps,
  IInputNumber,
  IInputNumberProps,
  IRadioGroup,
  IRadioGroupProps,
  ICustomProps,
  IRate,
  IRateProps,
  ISlider,
  ISliderProps,
  ISwitch,
  ISwitchProps,
  IListWrapProps,
  IListWrap,
  ITimePicker,
  ITimePickerProps,
  ITimeRangePicker,
  ITimeRangePickerProps,
  IUploadProps,
  IUpload,
  IInputGroup,
  IInputGroupProps,
  IAvatar,
  IAvatarProps,
  IImage,
  IImageProps,
  ITag,
  ITagProps,
  IProgress,
  IProgressProps,
  IFormItemGroup,
  IFormItemGroupProps,
  ICascader,
  ICascaderProps,
  ITreeSelect,
  ITreeSelectProps,
} from '../../typings';

export interface IBaseItem {
  /** 是否禁用 */
  disabled?: boolean;
  /** 只有custom类型 才有的渲染children方法 */
  render?: (props: ISchemaChildrenProps) => React.ReactNode;
  /** form.item 的包裹器。可以为item套一层外壳 */
  itemWraper?: () => React.ReactNode;
  /** 透传组件参数 */
  fieldProps?: unknown;
  /** col 的占位,如果item有。以item为结果 */
  col?: number;
  /** 默认：宽度100%。auto: 使用默认组件宽度 */
  width?: '100%' | 'auto';
}

type IUnionType =
  // 文本
  | ({
      type: IText;
    } & ITextProps)
  // 列表
  | ({
      type: IListWrap;
    } & IListWrapProps)
  // input
  | ({
      type: IInput;
    } & IInputProps)
  // input-group
  | ({
      type: IInputGroup;
    } & IInputGroupProps)
  // textarea
  | ({
      type: ITextArea;
    } & ITextAreaProps)
  // search
  | ({
      type: ISearch;
    } & ISearchProps)
  // password
  | ({
      type: IPassword;
    } & IPasswordProps)
  // intpu-number
  | ({
      type: IInputNumber;
    } & IInputNumberProps)
  // select
  | ({
      type: ISelect;
    } & ISelectProps)
  // datePicker
  | ({
      type: IDatePicker;
    } & IDatePickerProps)
  // rangePicker
  | ({
      type: IRangePicker;
    } & IRangePickerProps)
  // timePicker
  | ({
      type: ITimePicker;
    } & ITimePickerProps)
  // timeRangePicker
  | ({
      type: ITimeRangePicker;
    } & ITimeRangePickerProps)
  // checkbox
  | ({
      type: ICheckbox;
    } & ICheckboxProps)
  // checkbox-group
  | ({
      type: ICheckboxGroup;
    } & ICheckboxGroupProps)
  // radio-group
  | ({
      type: IRadioGroup;
    } & IRadioGroupProps)
  // rate
  | ({
      type: IRate;
    } & IRateProps)
  // slider
  | ({
      type: ISlider;
    } & ISliderProps)
  // switch
  | ({
      type: ISwitch;
    } & ISwitchProps)
  // upload
  | ({
      type: IUpload;
    } & IUploadProps)
  // Avatar
  | ({
      type: IAvatar;
    } & IAvatarProps)
  // Image
  | ({
      type: IImage;
    } & IImageProps)
  // Tag
  | ({
      type: ITag;
    } & ITagProps)
  // Progress
  | ({
      type: IProgress;
    } & IProgressProps)
  // ICascader
  | ({
      type: ICascader;
    } & ICascaderProps)
  // ITreeSelect
  | ({
      type: ITreeSelect;
    } & ITreeSelectProps)
  // 分组
  | ({
      type: IFormItemGroup;
    } & IFormItemGroupProps)
  // 自定义
  | ({
      type: ICustom;
    } & ICustomProps);
/** 渲染列表的单条数据类型 */
export type IFormItem = IBaseItem & IUnionType & FormItemProps;
/** 默认布局类型：Row */
export interface IRowLayoutType {
  type: 'row';
  /** 间距 默认:16 */
  gutter?: number;
  /** 间距 默认:6 */
  span?: number;
}
/** 自定义布局类型 */
export interface ICustomLayoutType {
  type: 'custom';
}
export type IUnionLayoutType = IRowLayoutType | ICustomLayoutType;
/** 表单的配置参数 */
export interface ISchemaBaseProps {
  /** 是否禁用，如果item有，以item的为结果 */
  disabled?: boolean;
  type: 'row' | 'custom';
  /** 布局类型 row: 使用默认的 Row 布局，custom: 可以增加group混合布局。 */
  /** 渲染列表 */
  options: IFormItem[];
  /** Form.useForm() */
  form?: FormInstance;
  initialValues?: Record<string, any>;
}
export type ISchemaProps = ISchemaBaseProps & IUnionLayoutType;
/** 透传给子组件children的参数: disabled|initialValues */
export type ISchemaChildrenProps = Pick<ISchemaBaseProps, 'disabled' | 'initialValues'> & {
  /** form的填充值 */
  value?: any;
  /** 自定义节点变更时，回调使用 */
  onChange?: (val?: any) => void;
  fieldProps?: any;
};
/** 需要继承的form参数 */
export type IFormInheritProps = {
  formDisabled?: boolean;
  initialValues?: any;
  values?: any;
};
/** item节点包裹器的参数 */
export type IItemWrapProps = IFormItem & IFormInheritProps;
