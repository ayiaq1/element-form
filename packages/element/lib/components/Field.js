/*
 * @Author: yihuang
 * @Date: 2021-11-30 18:27:39
 * @Description: 字段类型 包含 只读、编辑
 * @LastModifiedBy: yihuang
 */
var __assign =
  (this && this.__assign) ||
  function () {
    __assign =
      Object.assign ||
      function (t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
          s = arguments[i];
          for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
        }
        return t;
      };
    return __assign.apply(this, arguments);
  };
var __rest =
  (this && this.__rest) ||
  function (s, e) {
    var t = {};
    for (var p in s)
      if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0) t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === 'function')
      for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
        if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
          t[p[i]] = s[p[i]];
      }
    return t;
  };
import React, { useState, useEffect } from 'react';
import {
  Avatar,
  Checkbox,
  Divider,
  Image,
  Input,
  Progress,
  Radio,
  Rate,
  Slider,
  Switch,
  Tag,
  Upload,
} from 'antd';
import DateField from './DateField';
import Text from './Text';
import SelectField from './SelectField';
import TreeSelectField from './TreeSelectField';
import CascaderField from './CascaderField';
import DescriptionWrap from './DescriptionWrap';
function Field(_a) {
  var _b;
  var type = _a.type,
    render = _a.render,
    fieldProps = _a.fieldProps,
    resetProps = __rest(_a, ['type', 'render', 'fieldProps']);
  var _c = (_b = fieldProps) !== null && _b !== void 0 ? _b : {},
    children = _c.children,
    value = _c.value,
    resetFieldProps = __rest(_c, ['children', 'value']);
  var _d = useState(value !== null && value !== void 0 ? value : resetProps.value),
    val = _d[0],
    setVal = _d[1];
  useEffect(
    function () {
      setVal(value);
    },
    [value],
  );
  // 表单重置的时候值为 resetProps.value
  useEffect(
    function () {
      setVal(resetProps.value);
    },
    [resetProps.value],
  );
  switch (type) {
    case 'descriptions':
      return React.createElement(DescriptionWrap, { fieldProps: resetFieldProps });
    // 日期类型做一层数据转换
    case 'datePicker':
    case 'rangePicker':
    case 'timePicker':
    case 'timeRangePicker':
      return React.createElement(DateField, { fieldProps: fieldProps, value: val, type: type });
    case 'tag':
      return React.createElement(
        Tag,
        __assign({ color: 'default' }, fieldProps, { closable: false }),
        children
          ? children({
              value: val,
            })
          : val,
      );
    case 'avatar':
      return React.createElement(Avatar, __assign({}, resetFieldProps, { src: val }));
    case 'image':
      return React.createElement(
        Image,
        __assign({}, resetFieldProps, { preview: false, src: val }),
      );
    case 'checkbox':
      return React.createElement(
        Checkbox,
        __assign({}, resetFieldProps, { disabled: true, checked: val }),
      );
    case 'checkboxGroup':
      return React.createElement(
        Checkbox.Group,
        __assign({}, resetFieldProps, { disabled: true, value: val }),
      );
    case 'radioGroup':
      return React.createElement(
        Radio.Group,
        __assign({}, resetFieldProps, { disabled: true, value: val }),
      );
    case 'rate':
      return React.createElement(
        Rate,
        __assign({}, resetFieldProps, { disabled: true, value: val }),
      );
    case 'slider':
      return React.createElement(
        Slider,
        __assign({}, resetFieldProps, { disabled: true, value: val }),
      );
    case 'switch':
      return React.createElement(
        Switch,
        __assign({}, resetFieldProps, { disabled: true, checked: val }),
      );
    case 'progress':
      return React.createElement(Progress, __assign({}, resetFieldProps, { percent: val }));
    case 'upload':
      return React.createElement(
        Upload,
        __assign({}, resetFieldProps, {
          fileList: val,
          disabled: true,
          showUploadList: {
            showPreviewIcon: false,
            showRemoveIcon: false,
            showDownloadIcon: false,
          },
        }),
        children === null || children === void 0
          ? void 0
          : children(__assign(__assign({}, resetFieldProps), { value: val, disabled: true })),
      );
    case 'text':
    case 'textarea':
    case 'input':
    case 'search':
    case 'inputNumber':
    case 'inputGroup':
      return React.createElement(Text, { fieldProps: fieldProps, value: val });
    case 'password':
      return React.createElement(
        Input.Password,
        __assign({}, resetFieldProps, {
          visibilityToggle: false,
          bordered: false,
          disabled: true,
          value: val,
        }),
      );
    case 'select':
      return React.createElement(SelectField, { fieldProps: fieldProps, value: val });
    case 'treeSelect':
      return React.createElement(TreeSelectField, { fieldProps: fieldProps, value: val });
    case 'cascader':
      return React.createElement(CascaderField, { fieldProps: fieldProps, value: val });
    case 'divider':
      return React.createElement(Divider, __assign({}, resetFieldProps));
    case 'custom':
      return render === null || render === void 0
        ? void 0
        : render({
            value: val,
          });
    default:
      // 为以后做兼容 未识别的类型 也转换
      return React.createElement(
        React.Fragment,
        null,
        val === null || val === void 0 ? void 0 : val.toString(),
      );
  }
}
Field.displayName = 'Field';
export default Field;
