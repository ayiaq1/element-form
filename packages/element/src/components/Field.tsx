/*
 * @Author: yihuang
 * @Date: 2021-11-30 18:27:39
 * @Description: 字段类型 包含 只读、编辑
 * @LastModifiedBy: yihuang
 */

import React, { useState, useEffect } from 'react';
import {
  Avatar,
  Checkbox,
  Divider,
  Image,
  Input,
  Progress,
  Radio,
  Rate,
  Slider,
  Switch,
  Tag,
  Upload,
} from 'antd';
import type {
  ICustomProps,
  IDatePickerProps,
  IInputGroupProps,
  IInputNumberProps,
  IInputProps,
  IPasswordProps,
  IProps,
  IRangePickerProps,
  ISearchProps,
  ISelectFieldProps,
  ITagProps,
  ITextAreaProps,
  ITextProps,
  ITimePickerProps,
  ITimeRangePickerProps,
  ITreeSelectFieldProps,
  ICascaderFieldProps,
  IDividerProps,
  IDescriptionsProps,
} from '../typings';
import DateField from './DateField';
import Text from './Text';
import SelectField from './SelectField';
import TreeSelectField from './TreeSelectField';
import CascaderField from './CascaderField';
import DescriptionWrap from './DescriptionWrap';

/** descriptions */
function Field(props: IDescriptionsProps): JSX.Element;
/** divider */
function Field(props: IDividerProps): JSX.Element;
/** text */
function Field(props: ITextProps): JSX.Element;
/** input */
function Field(props: IInputProps): JSX.Element;
/** inputGroup */
function Field(props: IInputGroupProps): JSX.Element;
/** textarea */
function Field(props: ITextAreaProps): JSX.Element;
/** search */
function Field(props: ISearchProps): JSX.Element;
/** password */
function Field(props: IPasswordProps): JSX.Element;
/** InputNumber */
function Field(props: IInputNumberProps): JSX.Element;
/** datePicker */
function Field(props: IDatePickerProps): JSX.Element;
/** rangePicker */
function Field(props: IRangePickerProps): JSX.Element;
/** timePicker */
function Field(props: ITimePickerProps): JSX.Element;
/** TimeRangePicker */
function Field(props: ITimeRangePickerProps): JSX.Element;
/** Tag */
function Field(props: ITagProps): JSX.Element;
/** Custom */
function Field(props: ICustomProps): JSX.Element;
/** Select */
function Field(props: ISelectFieldProps): JSX.Element;
/** TreeSelect */
function Field(props: ITreeSelectFieldProps): JSX.Element;
/** Cascader */
function Field(props: ICascaderFieldProps): JSX.Element;
function Field({ type, render, fieldProps, ...resetProps }: IProps) {
  const { children, value, ...resetFieldProps } = (fieldProps as any) ?? {};
  const [val, setVal] = useState(value ?? resetProps.value);

  useEffect(() => {
    setVal(value);
  }, [value]);
  // 表单重置的时候值为 resetProps.value
  useEffect(() => {
    setVal(resetProps.value);
  }, [resetProps.value]);

  switch (type) {
    case 'descriptions':
      return <DescriptionWrap fieldProps={resetFieldProps} />;
    // 日期类型做一层数据转换
    case 'datePicker':
    case 'rangePicker':
    case 'timePicker':
    case 'timeRangePicker':
      return <DateField fieldProps={fieldProps as any} value={val} type={type} />;
    case 'tag':
      return (
        <Tag color="default" {...fieldProps} closable={false}>
          {children
            ? children({
                value: val,
              })
            : val}
        </Tag>
      );
    case 'avatar':
      return <Avatar {...resetFieldProps} src={val} />;
    case 'image':
      return <Image {...resetFieldProps} preview={false} src={val} />;
    case 'checkbox':
      return <Checkbox {...resetFieldProps} disabled={true} checked={val} />;
    case 'checkboxGroup':
      return <Checkbox.Group {...resetFieldProps} disabled={true} value={val} />;
    case 'radioGroup':
      return <Radio.Group {...resetFieldProps} disabled={true} value={val} />;
    case 'rate':
      return <Rate {...resetFieldProps} disabled={true} value={val} />;
    case 'slider':
      return <Slider {...resetFieldProps} disabled={true} value={val} />;
    case 'switch':
      return <Switch {...resetFieldProps} disabled={true} checked={val} />;
    case 'progress':
      return <Progress {...resetFieldProps} percent={val} />;
    case 'upload':
      return (
        <Upload
          {...resetFieldProps}
          fileList={val}
          disabled={true}
          showUploadList={{
            showPreviewIcon: false,
            showRemoveIcon: false,
            showDownloadIcon: false,
          }}
        >
          {children?.({
            ...resetFieldProps,
            value: val,
            disabled: true,
          })}
        </Upload>
      );
    case 'text':
    case 'textarea':
    case 'input':
    case 'search':
    case 'inputNumber':
    case 'inputGroup':
      return <Text fieldProps={fieldProps as any} value={val} />;
    case 'password':
      return (
        <Input.Password
          {...resetFieldProps}
          visibilityToggle={false}
          bordered={false}
          disabled={true}
          value={val}
        />
      );
    case 'select':
      return <SelectField fieldProps={fieldProps} value={val} />;
    case 'treeSelect':
      return <TreeSelectField fieldProps={fieldProps} value={val} />;
    case 'cascader':
      return <CascaderField fieldProps={fieldProps} value={val} />;
    case 'divider':
      return <Divider {...resetFieldProps} />;
    case 'custom':
      return render?.({
        value: val,
      });
    default:
      // 为以后做兼容 未识别的类型 也转换
      return <>{val?.toString()}</>;
  }
}
Field.displayName = 'Field';
export default Field;
