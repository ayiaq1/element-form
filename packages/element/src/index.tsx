/*
 * @Author: yihuang
 * @Date: 2021-10-09 11:46:52
 * @Description: 文本类型
 * @LastModifiedBy: yihuang
 */
import Element from './components/Element';
import SchemaForm from './components/SchemaForm';

export { Element, SchemaForm };

export default Element;
