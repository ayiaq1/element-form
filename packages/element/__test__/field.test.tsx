/*
 * @Author: yihuang
 * @Date: 2022-03-23 15:30:45
 * @Description: Field 组件
 * @LastModifiedBy: yihuang
 */
import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import moment from 'moment';
import mountTest from '../tests/mountTest';
import Field from '../src/components/Field';
import { EnumDataType } from '../src/typings';

const Time_YMD = '2022-01-14 02:06:04';
const Time_YMD_1 = '2022-06-27 13:29:13';
const HMS_1 = '02:06:04';
const HMS_2 = '13:29:13';
const unix = 1642097164;
// const unix_1 = 1656307753;
const unixValueOf = 1642097164000;
const unixValueOf_1 = 1656307753000;
const separator = '--';

/* Snapshot is a tool to test the code. */
describe('Field', () => {
  mountTest(Field);

  it('Save Field Text Snapshot', async () => {
    const wrapper = renderer.create(<Field type="text" value="text" />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('Save Field Text datePicker', async () => {
    const wrapper = renderer.create(<Field type="datePicker" />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('Save Field Text timePicker', async () => {
    const wrapper = renderer.create(<Field type="timePicker" />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('Save Field Text rangePicker', async () => {
    const wrapper = renderer.create(<Field type="rangePicker" />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('Save Field Text timeRangePicker', async () => {
    const wrapper = renderer.create(<Field type="timeRangePicker" />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  // ======== datePicker ==========
  it('Field type is datePicker default value', async () => {
    const wrapper = mount(<Field type="datePicker" value={Time_YMD} />);
    expect(wrapper.text()).toEqual(Time_YMD);
  });
  it('Field type is datePicker value to unixValueOf', async () => {
    const wrapper = mount(
      <Field
        type="datePicker"
        value={Time_YMD}
        fieldProps={{
          convertResult: 'unixValueOf',
        }}
      />,
    );
    expect(wrapper.text()).toEqual(unixValueOf.toString());
  });
  it('Field type is datePicker value to unix', async () => {
    const wrapper = mount(
      <Field
        type="datePicker"
        value={Time_YMD}
        fieldProps={{
          convertResult: 'unix',
        }}
      />,
    );
    expect(wrapper.text()).toEqual(unix.toString());
  });
  it('Field type is datePicker moment value', async () => {
    const wrapper = mount(<Field type="datePicker" value={moment(Time_YMD)} />);
    expect(wrapper.text()).toEqual(Time_YMD);
  });
  it('Field type is datePicker convertResult value', async () => {
    const wrapper = mount(
      <Field
        type="datePicker"
        value={Time_YMD}
        fieldProps={{
          convertResult: 'unixValueOf',
        }}
      />,
    );
    expect(wrapper.text()).toEqual(unixValueOf.toString());
  });
  // timePicker
  it('Field type is timePicker default value', async () => {
    const wrapper = mount(<Field type="timePicker" value={HMS_1} />);
    expect(wrapper.text()).toEqual(HMS_1);
  });
  it('Field type is timePicker moment value', async () => {
    const wrapper = mount(<Field type="timePicker" value={moment(HMS_1, EnumDataType.HMS)} />);
    expect(wrapper.text()).toEqual(HMS_1);
  });

  // rangePicker
  it('Field type is rangePicker default value', async () => {
    const wrapper = mount(<Field type="rangePicker" value={[Time_YMD, Time_YMD_1]} />);
    expect(wrapper.text()).toEqual(`${Time_YMD}-${Time_YMD_1}`);
  });
  it('Field type is rangePicker value to unixValueOf', async () => {
    const wrapper = mount(
      <Field
        type="rangePicker"
        value={[Time_YMD, Time_YMD_1]}
        fieldProps={{
          separator,
          convertResult: 'unixValueOf',
        }}
      />,
    );
    expect(wrapper.text()).toEqual(`${unixValueOf}${separator}${unixValueOf_1}`);
  });
  it('Field type is rangePicker moment value', async () => {
    const wrapper = mount(
      <Field
        type="rangePicker"
        value={[moment(Time_YMD), moment(Time_YMD_1)]}
        fieldProps={{
          separator,
          convertResult: 'unixValueOf',
        }}
      />,
    );
    expect(wrapper.text()).toEqual(`${unixValueOf}${separator}${unixValueOf_1}`);
  });

  // timeRangePicker
  it('Field type is timeRangePicker default value', async () => {
    const wrapper = mount(<Field type="timeRangePicker" value={[HMS_1, HMS_2]} />);
    expect(wrapper.text()).toEqual(`${HMS_1}-${HMS_2}`);
  });

  it('Field type is timeRangePicker moment value', async () => {
    const wrapper = mount(
      <Field
        type="timeRangePicker"
        value={[moment(HMS_1, EnumDataType.HMS), moment(HMS_2, EnumDataType.HMS)]}
        fieldProps={{
          separator,
        }}
      />,
    );
    expect(wrapper.text()).toEqual(`${HMS_1}${separator}${HMS_2}`);
  });

  // end
});
