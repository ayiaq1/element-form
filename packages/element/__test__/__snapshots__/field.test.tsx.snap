// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field Save Field Text Snapshot 1`] = `
<span
  aria-label="text"
  className="ant-typography ant-typography-ellipsis ant-typography-single-line"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "WebkitLineClamp": undefined,
    }
  }
>
  text
  <span
    aria-hidden={true}
    style={
      Object {
        "display": "block",
        "left": 0,
        "pointerEvents": "none",
        "position": "fixed",
        "top": 0,
        "visibility": "hidden",
        "whiteSpace": "nowrap",
        "wordBreak": "keep-all",
        "zIndex": -9999,
      }
    }
  >
    lg
  </span>
  <span
    aria-hidden={true}
    style={
      Object {
        "display": "block",
        "left": 0,
        "margin": 0,
        "padding": 0,
        "pointerEvents": "none",
        "position": "fixed",
        "top": 0,
        "visibility": "hidden",
        "whiteSpace": "normal",
        "width": 0,
        "zIndex": -9999,
      }
    }
  >
    <span
      aria-hidden={true}
    >
      ...
    </span>
  </span>
</span>
`;

exports[`Field Save Field Text datePicker 1`] = `""`;

exports[`Field Save Field Text rangePicker 1`] = `""`;

exports[`Field Save Field Text timePicker 1`] = `""`;

exports[`Field Save Field Text timeRangePicker 1`] = `""`;
